### PR TITLE
Writing code to the given package folder

### LINKED ISSUE #27: Make the Gradle task generate the code in the given package
This task depends on https://github.com/patxibocos/poetimizely/issues/26
Running the Gradle task should generate the code and write the files to the package.

### PR DESCRIPTION
**Pull request by patxibocos**
Fixes #27

### CODE REVIEW

**Comment by patxibocos** [comment]
I don't like adding too much code to the task as we cannot easily test/mock its dependencies.
But for now I think it is safe. We would figure out in the future if we really want/need an injector.

File: src/main/kotlin/com/patxi/poetimizely/plugin/GeneratorTask.kt [hunk 1]
```diff
@@ -18,6 +20,12 @@ open class GeneratorTask : DefaultTask() {
...
     fun doAction() {
+        val mainSrcDirs =
+            project.convention.getPlugin(JavaPluginConvention::class.java).sourceSets.getByName("main").allSource.srcDirs
+        val targetDir =
+            requireNotNull(mainSrcDirs.find { it.name == "kotlin" } ?: mainSrcDirs.find { it.name == "java" }) {
+                "Cannot find a source directory for Java/Kotlin"
+            }
         val optimizelyProjectId = optimizelyProjectId
         val optimizelyToken = optimizelyToken
         val packageName = packageName
```
Comment (ralphpina) [inline src/main/kotlin/com/patxi/poetimizely/plugin/GeneratorTask.kt:31]: Is this shadowing?

File: src/main/kotlin/com/patxi/poetimizely/plugin/GeneratorTask.kt [hunk 2]
```diff
@@ -9,7 +9,9 @@ import com.patxi.poetimizely.optimizely.buildExperimentsService
 import com.patxi.poetimizely.optimizely.buildFeaturesService
 import kotlinx.coroutines.runBlocking
 import org.gradle.api.DefaultTask
+import org.gradle.api.plugins.JavaPluginConvention
 import org.gradle.api.tasks.TaskAction
+import java.io.File
```
Comment (ralphpina) [inline src/main/kotlin/com/patxi/poetimizely/plugin/GeneratorTask.kt:14]: First Java code 😉

File: src/main/kotlin/com/patxi/poetimizely/plugin/GeneratorTask.kt (hunk 1 above)
Comment (patxibocos) [reply src/main/kotlin/com/patxi/poetimizely/plugin/GeneratorTask.kt:31]: Yes, I was feeling lazy about defining a meaningful name so I just shadowed them. Mutability cons you know...

**Pull request merged**